core/logic_events.go

```diff
@@ -264,6 +264,14 @@ func (e eventsLogic) processWebToolsEvents() {
 		return
 	}
 
+	err = e.preProcessLocation(allWebToolsEvents)
```

Comment:
The corresponding function for the images was called processImages but you introduce preProcessLocation. Please keep the same naming convetions.